src/headers.rs

```diff
@@ -1,17 +1,78 @@
-pub enum Header {
+use http::request::Builder;
+use std::collections::HashMap;
+
+pub enum HeaderField {
     User(String),
     Source(String),
     Catalog(String),
     Schema(String),
     // TimeZone,  // Not yet supported
     // Language,  // Not yet supported
     TraceToken(String),
-    Session(String, String),
+    Session(HashMap<String, String>),
     // Role,  // Not yet supported
     // PreparedStatement,  // Not yet sypported
     TransactionId(String),
     ClientInfo(String),
     ClientTag(String),
     // ResourceEstimate, // Not yet supported
     // ExtraCredential, // Not yet supported
-}
\ No newline at end of file
+}
+
+impl HeaderField {
+    fn get_key(&self) -> String {
```

Comment:
Here's a little magic.  Change this to

```
fn get_key(&self) -> &'static str {
  match self {
    HeaderField::User(_) => "User",
    ...
  }
}
```

Since we are just using it as a reference below, this will avoid an allocation.  Literal strings are stored in the static section, so they are available for the lifetime of the program.  They are effectively string constants.  Then below, we can use them slightly differently... (see below)